src/Checkers/LicenseChecker.js

```diff
@@ -0,0 +1,329 @@
+     */
+    _skipToNextLine(text, pos) {
+        while (pos < text.length && text[pos] != '\n') {
+            pos++;
+        }
+        if (pos != text.length) pos++;
+        return pos;
+    }
+
+    _createJsonResponse(token, lineNum, linePos, currentPos) {
+        return {
+            token: token,
+            lineNum: lineNum,
+            linePos: currentPos - linePos
+        };
+    }
+
+    /**
+     * Skips to the text after the single line comments ("//") starting from _pos.
+     * @private
+     * @return offset from the current position (_pos) to the commented text or null if no comments found.
+     */
+    _skipToTextInComments(text, startPos) {
+        if (startPos + 1 >= text.length) return null;
+        if (text[startPos] != '/' || text[startPos + 1] != '/') {
+            return null;
+        } else {
+            return startPos + 2;
+        }
+    }
+
+    /**
+     * Check path for License mistakes
+     * @param {String} path
+     * @return {[ErrorMessage]}
+     */
+    check(dirpath) {
+        const allFiles = this._getFiles(dirpath, []);
+        const errors = [];
+        errors.push(false); // reserved for LICENSE error
+        for (const file of allFiles) {
+            const parsedPath = path.parse(file);
+            if (this._extensionSet.has(parsedPath.ext)) {
+                errors.push(this._compareWithLicense(file, true));
+            } else if (parsedPath.name === 'LICENSE') {
+                errors[0] = this._compareWithLicense(file, false); // assign to reserved place
+            }
+        }
+        return errors.filter((error) => error != false);
+    }
+
+    /**
+     * Check path for License mistakes
+     * @param {String} filepath path to file
+     * @param {Boolean} parseSourceComments true if the license text
+     * is supposed to be commented out (source file header) or uncommented (LICENSE file)
+     * @private
+     * @return {ErrorMessage}
+     */
+    _compareWithLicense(filepath, parseSourceComments) {
+        const yearRegexp = /\d\d\d\d(\-\d\d\d\d)?/;
+
+        let testedLicense = fs.readFileSync(filepath, 'utf-8');
+        testedLicense = this._delete3dPartyCopyright(testedLicense).replace(yearRegexp, '');
+        const goldenLicense = fs.readFileSync(LICENSE_FILE_PATH, 'utf-8');
+
+        const output = this._compareTwoLicenseTexts(testedLicense, goldenLicense, parseSourceComments);
+        return output ? new ErrorMessage(output.message, output.lineNum, filepath, 'LicenseChecker', output.linePos) : output;
+    }
+
+    /**
+     * Delete third-party copyright in license
+     * @param {String} license text of license
+     * @private
+     * @return {String} new license
+     */
+    _delete3dPartyCopyright(license) {
+        /* regexp match example:
+         * "// Copyright 2017 company name\n"
+         * "Copyright 2016 companyName\r\n"
+         */
+        const copyrightStringRegexp = /(\/\/\s)?Copyright\s\d\d\d\d(\-\d\d\d\d)?\s(.*)(\r)?\n/g;
```

Comment:
This regexp is wrong because:

1. there may be multiple spaces between the words here, a single space shouldn't be enforced
2. a space(s) is(are) optional after //
3. there may be no Electric Imp copyright header over there